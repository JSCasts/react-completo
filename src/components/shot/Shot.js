import React, { Component } from 'react';

import ShotDescription from '../shot-description/ShotDescription';
import parseDate from '../../services/date';
import styles from './styles.scss';

class Shot extends Component {
  constructor(props) {
    super(props);

    this.state = {
      showDescription: false,
    };

    this.toggle = this.toggle.bind(this);
  }

  toggle() {
    this.setState({ showDescription: !this.state.showDescription });
  }

  render() {
    const { id, image, date, title, description, likesCount, viewsCount } = this.props;

    return (
      <div onMouseEnter={this.toggle} onMouseLeave={this.toggle} className={styles.container}>
        <div className={styles.content}>
          <img alt="Shot" src={image} />
          <ShotDescription
            id={id}
            showDescription={this.state.showDescription}
            description={description}
            title={title}
          />
        </div>
        <div className={styles.tools}>
          <span className={styles.calendar}>{parseDate(date)}</span>
          <span className={styles.views}>{viewsCount}</span>
          <span className={styles.likes}>{likesCount}</span>
        </div>
      </div>
    );
  }
}

Shot.defaultProps = {
  description: '',
};

Shot.propTypes = {
  id: React.PropTypes.number.isRequired,
  image: React.PropTypes.string.isRequired,
  viewsCount: React.PropTypes.number.isRequired,
  likesCount: React.PropTypes.number.isRequired,
  date: React.PropTypes.string.isRequired,
  title: React.PropTypes.string.isRequired,
  description: React.PropTypes.string,
};

export default Shot;
