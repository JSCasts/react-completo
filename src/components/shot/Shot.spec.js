import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Shot from './Shot';

const props = {
  id: 1,
  image: 'image',
  viewsCount: 17,
  likesCount: 23,
  date: '2010-10-10',
  title: 'title',
  description: 'description',
};

test('Snapshot', () => {
  const component = renderer.create(
    <Shot {...props} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onMouseEnter();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onMouseLeave();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Shot', () => {
  test('initial state', () => {
    const component = shallow(
      <Shot {...props} />
    );
    expect(component.state()).toEqual({ showDescription: false });
  });

  test('has texts', () => {
    const component = shallow(
      <Shot {...props} />
    );
    expect(component.text()).toContain('ShotDescription');
    expect(component.text()).toContain('17');
    expect(component.text()).toContain('23');
    expect(component.text()).toContain('2010-10-10');
  });

  test('toggle', () => {
    const component = shallow(
      <Shot {...props} />
    );
    component.instance().toggle();
    expect(component.state()).toEqual({ showDescription: true });

    component.instance().toggle();
    expect(component.state()).toEqual({ showDescription: false });
  });
});
