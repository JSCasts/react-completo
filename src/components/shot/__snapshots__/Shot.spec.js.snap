exports[`test Snapshot 1`] = `
<div
  className="container"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  <div
    className="content">
    <img
      alt="Shot"
      src="image" />
  </div>
  <div
    className="tools">
    <span
      className="calendar">
      2010-10-10
    </span>
    <span
      className="views">
      17
    </span>
    <span
      className="likes">
      23
    </span>
  </div>
</div>
`;

exports[`test Snapshot 2`] = `
<div
  className="container"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  <div
    className="content">
    <img
      alt="Shot"
      src="image" />
    <a
      onClick={[Function]}
      style={Object {}}>
      <div
        className="container">
        <h4>
          title
        </h4>
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "description",
            }
          } />
      </div>
    </a>
  </div>
  <div
    className="tools">
    <span
      className="calendar">
      2010-10-10
    </span>
    <span
      className="views">
      17
    </span>
    <span
      className="likes">
      23
    </span>
  </div>
</div>
`;

exports[`test Snapshot 3`] = `
<div
  className="container"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  <div
    className="content">
    <img
      alt="Shot"
      src="image" />
  </div>
  <div
    className="tools">
    <span
      className="calendar">
      2010-10-10
    </span>
    <span
      className="views">
      17
    </span>
    <span
      className="likes">
      23
    </span>
  </div>
</div>
`;
